src/osd/PrimaryLogPG.cc

```diff
@@ -4449,8 +4547,10 @@ int PrimaryLogPG::trim_object(
     ctx->delta_stats.num_object_clones--;
     if (coi.is_cache_pinned())
       ctx->delta_stats.num_objects_pinned--;
-    if (coi.has_manifest())
+    if (coi.has_manifest()) {
+      dec_all_refcount_head_manifest(head_obc->obs.oi, ctx.get(), coid);
```

Comment:
I don't think this is right.  The oi you are passing here is used for the manifest, so you should be passing the object_info_t for the clone, not for head.  If I'm right about this, please ensure that the fix and the test case are easy to find for the next round of reviews.